test/spec/LoginRouter_spec.js

```diff
@@ -533,6 +533,20 @@ function (Okta, Q, Backbone, xdomain, SharedUtil, OktaAuth, Util, Expect, Router
       });
     });
 
+    describe('Trigger event is called', function () {
+      itp('Event is triggerrd on router start', function () {
+        return setup()
+            .then(function (test) {
+              test.router.primaryAuth();
+              spyOn(test.router.controller, 'triggerEvent');
```

Comment:
Some comments:

1) Better names for describe/test:
```javascript
// Describe is a general subheader type name
describe('Events', function () {
  // it statements should be read as if you start with "it", i.e. "it triggers a ..."
  // these two are necessary because we go down two different code paths
  it('triggers a pageRendered event when first controller is loaded');
  it('triggers a pageRendered event when navigating to a new controller');
});
```

2) Better to test "functionality", vs. testing that "function was called", i.e.:
```javascript

// In setup, create a spy that we can check against
    function setup(settings) {
      // stuff
      var eventSpy = new jasmine.createSpy('eventSpy');
      var router = new Router(_.extend({
        // stuff
      }, settings));
      router.on('all', eventSpy);
      return tick().then(function () {
        return {
           // stuff
           eventSpy: eventSpy
        };
      });
    }

// And then in your test function you can check that the spy was called
```

3) You should actually test what event was triggered
```javascript
// Make sure we're not sending too many events
expect(test.eventSpy.calls.count()).toBe(1);

// Expect that we got the event that we're expecting
expect(test.eventSpy).toHaveBeenCalledWith('pageRendered', { page: 'primary-auth' );
```